examples/hello_world/src/lib.rs

```diff
@@ -25,45 +25,66 @@ mod proto;
 
 use oak_derive::OakNode;
 use proto::hello_world::{HelloRequest, HelloResponse};
-use protobuf::Message;
-use std::io::Write;
+use proto::hello_world_grpc::{dispatch, HelloWorldNode};
 
 #[derive(OakNode)]
 struct Node;
 
-// TODO: Generate this code via a macro or code generation (e.g. a protoc plugin).
-trait HelloWorldNode {
-    fn say_hello(&self, req: &HelloRequest) -> HelloResponse;
-}
-
-// TODO: Generate this code via a macro or code generation (e.g. a protoc plugin).
 impl oak::Node for Node {
     fn new() -> Self {
         oak_log::init(log::Level::Debug).unwrap();
         Node
     }
     fn invoke(&mut self, grpc_method_name: &str, grpc_channel: &mut oak::Channel) {
-        let mut logging_channel = oak::logging_channel();
-        match grpc_method_name {
-            "/oak.examples.hello_world.HelloWorld/SayHello" => {
-                let req = protobuf::parse_from_reader(grpc_channel).unwrap();
-                let res = (self as &mut HelloWorldNode).say_hello(&req);
-                res.write_to_writer(grpc_channel).unwrap();
-            }
-            _ => {
-                writeln!(logging_channel, "unknown method name: {}", grpc_method_name).unwrap();
-                panic!("unknown method name");
-            }
-        };
+        dispatch(self as &mut HelloWorldNode, grpc_method_name, grpc_channel)
     }
 }
 
-// TODO: Generate parts of this code via a macro or code generation (e.g. a protoc plugin).
 impl HelloWorldNode for Node {
-    fn say_hello(&self, req: &HelloRequest) -> HelloResponse {
-        let mut res = HelloResponse::new();
+    fn say_hello(&mut self, req: HelloRequest) -> HelloResponse {
         info!("Say hello to {}", req.greeting);
+        let mut res = HelloResponse::new();
         res.reply = format!("HELLO {}!", req.greeting);
         res
     }
+
+    fn lots_of_replies(&mut self, req: HelloRequest) -> Vec<HelloResponse> {
```

Comment:
I understand that the current streaming stuff is temporary, but using `Vec` is not going to work, we need callbacks or channels of some sort to have true streaming.